pkg/controllers/externalsecret/externalsecret_controller_test.go

```diff
@@ -112,11 +119,93 @@ var _ = Describe("ExternalSecret controller", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 
+		It("should increment the syncCallsTotal metric", func() {
```

Comment:
Adding conditions that the metrics properly changed to some of the existing tests seems reasonable, but specific test cases for metrics has code smell to me. Test cases should follow from the various possible code paths in the system, metrics are a possible output from paths within controller, but standalone tests cases for metric seem fragile.